File: app/src/main/java/com/github/clebermatheus/yahooweather/activities/MainActivity.java
package com.github.clebermatheus.yahooweather.activities;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.TextView;

import com.github.clebermatheus.yahooweather.R;
import com.github.clebermatheus.yahooweather.models.Channel;
import com.github.clebermatheus.yahooweather.models.Item;

import org.json.JSONObject;

import static android.os.Build.VERSION.SDK_INT;
import static android.os.Build.VERSION_CODES.N;
import static android.text.Html.FROM_HTML_OPTION_USE_CSS_COLORS;
import static android.text.Html.fromHtml;
import static com.github.clebermatheus.yahooweather.utils.Util.PREFERENCES;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        SharedPreferences sharedPreferences = getSharedPreferences(PREFERENCES, MODE_PRIVATE);
        String resultJSON = sharedPreferences.getString("resultJSON", "");
        Log.i(TAG, resultJSON);
        exibeClimaAtual(resultJSON);
    }

    public void exibeClimaAtual(String jsonString){
        try{
            JSONObject obj = new JSONObject(jsonString);
            JSONObject results = obj.getJSONObject("query").getJSONObject("results");
            Channel channel = new Channel(results.getJSONObject("channel"));
            Item item = channel.getItem();
            ActionBar actionBar = getSupportActionBar();
            TextView conditionToday = (TextView) findViewById(R.id.conditionToday);
            Item.Condition condition = item.getCondition();
            Channel.Location location = channel.getLocation();
            conditionToday.setText(condition.getTemp()+" º"+channel.getUnits().getTemperature()+" em "+
                    location.getCity()+" com tempo "+condition.getCode()+".");
            if (actionBar != null) {actionBar.setTitle(channel.getTitle());}
        }catch(Exception e){e.printStackTrace();}
    }
}

File: app/src/main/java/com/github/clebermatheus/yahooweather/activities/MainActivity.kt
package com.github.clebermatheus.yahooweather.activities

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.TextView
import com.github.clebermatheus.yahooweather.R
import com.github.clebermatheus.yahooweather.models.Channel
import com.github.clebermatheus.yahooweather.models.Item
import com.github.clebermatheus.yahooweather.models.ResultQuery
import com.github.clebermatheus.yahooweather.utils.Util.PREFERENCES
import com.github.clebermatheus.yahooweather.utils.Util.getWeatherCode
import com.google.gson.Gson

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val sharedPreferences = getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE)
        val resultJSON = sharedPreferences.getString("resultJSON", "")
        Log.i(TAG, resultJSON)
        exibeClimaAtual(resultJSON)
    }

    private fun exibeClimaAtual(json: String){
        val actionBar = supportActionBar
        val gson = Gson()
        val resultado: ResultQuery = gson.fromJson(json, ResultQuery::class.java)
        Log.d(TAG, resultado.toString())
        val channel: Channel = resultado.query!!.results!!.channel ?: Channel()
        val conditionToday = findViewById<TextView>(R.id.conditionToday)

        // Condition Today
        val (code, _, temp, _) = channel.item?.condition ?: Item.Condition()
        val city = channel.location?.city ?: ""
        val temperature = channel.units?.temperature ?: ""
        conditionToday.text = "${temp}º${temperature} em ${city} com tempo ${code.getWeatherCode()}."

        if (actionBar != null) {
            actionBar.title = channel.title
        }else{}
    }

    companion object {
        private val TAG = "MainActivity"
    }
}
